### ISSUE TITLE
Error if no warnings are returned by dialyzer

### ISSUE THREAD

**Issue by chrismcg** [Comcast/dialyzex#7]
I turned off `underspecs` to remove the noise from internal Elixir warnings on a new project and it causes the following error:

```
chris@anez % mix dialyzer --no-check
Running analysis...
** (FunctionClauseError) no function clause matching in Mix.Tasks.Dialyzer.format_results/1

    The following arguments were given to Mix.Tasks.Dialyzer.format_results/1:

        # 1
        []

    Attempted function clauses (showing 4 out of 4):

        defp format_results({[], []})
        defp format_results({ignored, []})
        defp format_results({[], failed})
        defp format_results({ignored, failed})

    lib/mix/tasks/dialyzer.ex:186: Mix.Tasks.Dialyzer.format_results/1
    lib/mix/tasks/dialyzer.ex:182: Mix.Tasks.Dialyzer.run/1
    (mix) lib/mix/task.ex:314: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:80: Mix.CLI.run_task/2
    (elixir) lib/code.ex:677: Code.require_file/2
```

I've put up a sample repo here that illustrates the issue:

https://github.com/chrismcg/diatest

**Comment by seancribbs** [comment]
Oh, the irony! This is the sort of bug dialyzer is supposed to find. Luckily I believe the fix is simple, namely making the first clause of `report_warnings` return the same type as the second one. 

If it's ok with you, I'm going to create the first "test" based on your sample code.

**Comment by seancribbs** [comment]
This was partly resolved by #8, will add tests later.
